api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -87,6 +108,9 @@ type VolumeReplicationGroupSpec struct {
 	// List of unique S3 profiles in RamenConfig that should be used to store
 	// and forward PV related cluster state to peer DR clusters.
 	S3Profiles []string `json:"s3Profiles"`
+
+	// Indiciation to VRG about whether it is MetroDR or RegionalDR.
+	Mode Mode `json:"mode"`
```

Comment:
One big comment:

- I think we should not mix up fencing with VRG at all, fencing should be a separate CRD and it should be managed outside the VRG scope
    - For example each VRG should not attempt to fence or unfence a cluster, it should be controlled more centrally by its own CRD
- The spec changes I would see/think are like so based on fencing being out of scope of VRG CRD as such:
    ```
    spec:
      pvcSelector:
        matchLabels:
          any-pvc-label: value
      s3ProfileNames:
        - s3-profile-of-east
        - s3-profile-of-west
      modes:
        - sync
        - async
      async:
        schedulingInterval: "1h"
        replicationState: "Primary"
        replicationSelector:
          matchLabels:
            class: ramen
    ```

    - The `modes` is a list so as to be extensible in the future
    - `async` is for async specific parameters only
    - There is no corresponding `sync` section as I am unsure if we have any further parameters that we need for the same

- Fencing is something that can be optionally checked when a VRG is created for correctness, but as this could be a user controlled CRD, pointing to a fencing resource can be ignored when a user creates the CR and may need appropriate handling.
    - For example add spec.sync.fenceRef (v1.ObjectReference), where the fenceRef is a cluster scoped resource to ensure that local cluster is not fenced and hence workload can be run on the local cluster (do/can we extend it to remote cluster as well? How do we know it is a local cluster?)
    - I guess more questions than answers here, hence could be optional and a future consideration

In all VRG acts to restore PVs (and PVCs as we extend the backup to the same) when in sync mode, and does not really do anything more at present.

Thoughts?